chainerio/__init__.py

```diff
@@ -68,60 +87,194 @@ def open(file_path: str, mode: str = 'rb',
+    Otherwise, it will be set to default,
+    which represents the default working directory as follow:
+
+    +---------+---------------------------+
+    |         |     Default Root Path     |
+    +=========+===========================+
+    | POSIX   | current working directory |
+    +---------+---------------------------+
+    | HDFS    |     /user/``USERNAME`` #1 |
+    +---------+---------------------------+
+    | zip     |      top directory        |
+    +---------+---------------------------+
+
+    #1 For the details about the ``USERNAME`` in HDFS, please refer to
+    `HDFS Document <https://hadoop.apache.org/docs/current/\
+            hadoop-project-dist/hadoop-hdfs/HdfsPermissionsGuide.html>`_
+
+
+    Args:
+        uri_or_handler (str or :class:~`chainerio.IO`): The ``uri_or_handler``
+            can accept the following three kinds of values:
+
+            1. the name of a handler (string): set the default handler
+                to the corresponding handler, and root path.
+                See :func:`create_handler` for supported name of handlers.
+
+            2. an uri of directory (string): set the context
+                to use the corresponding handler and set
+                the root path to the given directory.
+
+            3. a handler, which is an instance of
+                :class:`chainerio.IO`. Set the default handler
+                to the given handler, and root path to default.
+
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     default_context.set_root(uri_or_handler)
 
 
-def create_handler(uri: str) -> IO:
+def create_handler(handler_name: str) -> IO:
+    """Returns a handler according to the given ``handler_name``
+
+    The current supported handlers are:
+
+    1. posix
+    2. hdfs
+
+    Args:
+        handler_name (str): the name of handler to create
+
+    Returns:
+        An object of :class:`chainerio.IO`
+        according to the ``handler_name``
+
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     (handler, actual_path, is_URI) = \
-        default_context.get_handler_by_name(uri)
+        default_context.get_handler_by_name(handler_name)
     return handler
 
 
-def isdir(file_path: str) -> bool:
+def isdir(path: str) -> bool:
+    """Returns ``Ture`` if the path is an existing directory
+
+    Forward to the corresponding :func:`IO.isdir` upon the default handler.
+    Refer to the link for more details.
+
+    The ``path`` can be  an Unix path or an URI.
+    """
     global _DEFAULT_CONTEXT
     default_context = _DEFAULT_CONTEXT
 
     (handler, actual_path, is_URI) = \
-        default_context.get_handler_by_name(file_path)
+        default_context.get_handler_by_name(path)
     return handler.isdir(actual_path)
 
 
-def mkdir(file_path: str, mode: int = 0o777, **kwargs) -> None:
+def mkdir(path: str, mode: int = 0o777, *,
+          dir_fd: Optional[int] = None) -> None:
```

Comment:
How the `dir_fd` works in HDFS and potentially in ZIP? Although our principal is to align with POSIX, this is too specific and hard to export to other systems.

Reply:
They simply ignored, like lots of the variables in the `open`. We keep them in case they are given.

Reply:
My thought is this should be removed, but changing API is out of scope so far.